### PR TITLE
Smc105section

### PR DESCRIPTION
**Pull request by ningosi**
@ssmusoke find time to look into this

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -103,11 +103,11 @@
 		</dependency>
 
 		<dependency>
- 	         <groupId>org.openmrs.module</groupId>
```
Comment (ssmusoke) [inline api/pom.xml]: @ningosi No changes in this file

File: api/src/main/java/org/openmrs/module/ugandaemrreports/data/converter/FaciltyAndOutReachDataConverter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.module.reporting.data.converter.DataConverter;
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+
+/**
+ * Created by Nicholas Ingosi on 5/22/17.
+ */
+public class FaciltyAndOutReachDataConverter implements DataConverter {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/data/converter/FaciltyAndOutReachDataConverter.java]: @ningosi Prefix SMC on this class

File: api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/CustomParametersDatasetEvaluator.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.openmrs.module.ugandaemrreports.definition.dataset.definition.GlobalPropertyParametersDatasetDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+/**
+ * Created by codehub on 7/19/17.
+ */
+@Handler(supports = GlobalPropertyParametersDatasetDefinition.class)
+public class CustomParametersDatasetEvaluator implements DataSetEvaluator {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/CustomParametersDatasetEvaluator.java]: @ningosi This was to be changed to GlobalPropertyDatasetEvaluator in another PR

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/CommonCohortDefinitionLibrary.java [hunk 4]
```diff
@@ -23,6 +19,11 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
+import java.util.Date;
+import java.util.UUID;
+
+
 /**
  * Library of common Cohort definitions
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/CommonCohortDefinitionLibrary.java]: @ningosi Not sure what changes are in this file - should be left untouched

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 5]
```diff
@@ -28,16 +23,20 @@
...
 import org.openmrs.module.ugandaemrreports.reporting.utils.ReportUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
+import java.util.Date;
+
 /**
- * Cohorts for the MoH 105 report
+ * Created by Nicholas Ingosi on 5/23/17.
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ningosi Remove the created by and use the previous overwritten class comment

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 6]
```diff
@@ -57,8 +56,8 @@ public CohortDefinition femaleAndHasAncVisit(double lower, double upper){
         cd.addParameter(new Parameter("onOrAfter", "Start Date", Date.class));
         cd.addParameter(new Parameter("onOrBefore", "End Date", Date.class));
         cd.addSearch("female", ReportUtils.map(definitionLibrary.females(), ""));
-        cd.addSearch("ancVist", ReportUtils
-		        .map(totalAncVisits(lower, upper), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore}"));
+        cd.addSearch("ancVist", ReportUtils.map(totalAncVisits(lower, upper), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore}"));
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ningosi Why these formatting changes

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 7]
```diff
@@ -178,4 +171,123 @@ public CohortDefinition missedANCAppointment() {
...
         return cd;
     }
+
+    //libraries for the SMC section follow here
+
+    /**
+     *
+     * @return CohortDefinition
+     */
+    public CohortDefinition siteTypeFacilitySc(){
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ningosi Global - need to limit the cohorts to SMC Encounter

Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ningosi Write SC suffix in full

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 8]
```diff
@@ -178,4 +171,123 @@ public CohortDefinition missedANCAppointment() {
...
+
+        return cd;
+
+    }
+
+    /**
+     *
+     * @return CohortDefinition
+     */
+    public CohortDefinition siteTypeFacilityDc(){
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ningosi Write DC suffix in full

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/calculation/smc/SMCAdrressCalculation.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.AbstractPatientCalculation;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Created by Nicholas Ingosi  on 5/22/17.
+ */
+public class SMCAdrressCalculation extends AbstractPatientCalculation {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/calculation/smc/SMCAdrressCalculation.java]: @ningosi What makes the SMC address calculation different from other address calculations

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/calculation/smc/SmcReturnFollowUpCalculation.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+import org.openmrs.module.ugandaemrreports.reporting.calculation.Calculations;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.EmrCalculationUtils;
+
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * Created by Nicholas Ingosi on 8/1/17.
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/calculation/smc/SmcReturnFollowUpCalculation.java]: @ningosi Global remove the created by Nicholas ... This may be a separate PR

**Comment by ssmusoke** [comment]
@ningosi This PR includes alot of formatting changes not required in it, also there are conflicting files committed

**Comment by ningosi** [comment]
I will have to redo this a fresh. sorry

**Comment by ningosi** [comment]
Closing this PR. It has been redone in https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/pull/64 @ssmusoke for your action
